src/main.js

```diff
@@ -244,3 +244,81 @@ let destinations = [
   ]
 
 
+  //visited and bucktelist lists
+  const visited = destinations.filter(item => item.visited === true);
+  const bucketlist = destinations.filter(item => item.visited === false);
+ 
+
+  //get list of urls
+  const getURL = (destinations) => {
+    let url= destinations.map(obj => obj.photo.valueOf());
+    return url;
+  }
+
+  //create elements for each city
+  const elements= destinations.forEach(destination => {
+
+    //picture element
+    const createImg = (url) => {
+      let figure = document.createElement('figure');
+    //img element
+      let img = document.createElement('img');
+      img.style.objectFit ='cover';
+      document.body.appendChild(figure);
+      figure.appendChild(img);
+    //img caption
+      let figcaption= document.createElement('figcaption');
+      figure.appendChild(figcaption);
+      figcaption.innerText= `${destinations.city}, ${destinations.country}`
+
+
+      //get urls
+      for(let i=0; i<destinations.length; i++) {
+        img.src = url[i];
+      };
+      
+      //add classes 
+      if(destination.visited === true) {
+        img.classList.add('visited');
+      }
+      else {
+        img.classList.add('bucketlist');
+      }
+
+  };
+
+
+
+  return createImg(getURL(destinations));
+
+});
+
+  //create buttons
+  const buttonVisited = document.createElement('button');
+  buttonVisited.innerText='Visited';
+  buttonVisited.type='button';
+  document.body.appendChild(buttonVisited);
+  
+  
+  const buttonBucketlist = document.createElement('button');
+  buttonBucketlist.innerText='Bucketlist';
+  buttonBucketlist.type='button';
+  document.body.appendChild(buttonBucketlist);
```

Comment:
These buttons are not dynamic like the images, they could be done directly on HTML side ;)